src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -7,6 +7,8 @@
 #include "ofi_impl.h"
 #include "ofi_events.h"
 
+#define MPIDI_OFI_MR_KEY_PREFIX_SHIFT 63
+
```

Comment:
Perhaps we want to add two notes to `MPIDI_OFI_mr_key_alloc` that:
(1) we use the highest bit of every key to distinguish autogenerated key with the user-specified key.
(2) because the highest bit of key is reserved to indicate the key type, a vaild key can be at most 63 bits, including the user-specified `requested_key`.

Reply:
ok